metricbeat/helper/prometheus/textparse.go

```diff
@@ -0,0 +1,694 @@
+	}
+	return 0
+}
+
+type Histogram struct {
+	SampleCount      *uint64
+	SampleSum        *float64
+	Bucket           []*Bucket
+	IsGaugeHistogram bool
+}
+
+func (m *Histogram) GetSampleCount() uint64 {
+	if m != nil && m.SampleCount != nil {
+		return *m.SampleCount
+	}
+	return 0
+}
+
+func (m *Histogram) GetSampleSum() float64 {
+	if m != nil && m.SampleSum != nil {
+		return *m.SampleSum
+	}
+	return 0
+}
+
+func (m *Histogram) GetBucket() []*Bucket {
+	if m != nil {
+		return m.Bucket
+	}
+	return nil
+}
+
+type OpenMetric struct {
+	Label       []*labels.Label
+	Exemplar    *exemplar.Exemplar
+	Name        *string
+	Gauge       *Gauge
+	Counter     *Counter
+	Info        *Info
+	Stateset    *Stateset
+	Summary     *Summary
+	Unknown     *Unknown
+	Histogram   *Histogram
+	TimestampMs *int64
+}
+
+func (m *OpenMetric) GetName() *string {
+	if m != nil {
+		return m.Name
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetLabel() []*labels.Label {
+	if m != nil {
+		return m.Label
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetGauge() *Gauge {
+	if m != nil {
+		return m.Gauge
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetCounter() *Counter {
+	if m != nil {
+		return m.Counter
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetInfo() *Info {
+	if m != nil {
+		return m.Info
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetStateset() *Stateset {
+	if m != nil {
+		return m.Stateset
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetSummary() *Summary {
+	if m != nil {
+		return m.Summary
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetUnknown() *Unknown {
+	if m != nil {
+		return m.Unknown
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetHistogram() *Histogram {
+	if m != nil && m.Histogram != nil && !m.Histogram.IsGaugeHistogram {
+		return m.Histogram
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetGaugeHistogram() *Histogram {
+	if m != nil && m.Histogram != nil && m.Histogram.IsGaugeHistogram {
+		return m.Histogram
+	}
+	return nil
+}
+
+func (m *OpenMetric) GetTimestampMs() int64 {
+	if m != nil && m.TimestampMs != nil {
+		return *m.TimestampMs
+	}
+	return 0
+}
+
+type MetricFamily struct {
+	Name   *string
+	Help   *string
+	Type   textparse.MetricType
+	Unit   *string
+	Metric []*OpenMetric
+}
+
+func (m *MetricFamily) GetName() string {
+	if m != nil && m.Name != nil {
+		return *m.Name
+	}
+	return ""
+}
+func (m *MetricFamily) GetUnit() string {
+	if m != nil && *m.Unit != "" {
+		return *m.Unit
+	}
+	return ""
+}
+
+func (m *MetricFamily) GetMetric() []*OpenMetric {
+	if m != nil {
+		return m.Metric
+	}
+	return nil
+}
+
+const (
+	//suffixInfo   = "_info"
+	suffixTotal  = "_total"
+	suffixGCount = "_gcount"
+	suffixGSum   = "_gsum"
+	suffixCount  = "_count"
+	suffixSum    = "_sum"
+	suffixBucket = "_bucket"
+)
+
+//func isInfo(name string) bool {
+//	return len(name) > 5 && name[len(name)-5:] == suffixInfo
+//}
+
+// Counters have _total suffix
+func isTotal(name string) bool {
+	return len(name) > 6 && name[len(name)-6:] == suffixTotal
```

Comment:
Nit. I think you can trust on the stdlib and the compiler for these optimizations, and the code would be more clear here :slightly_smiling_face: https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/strings/strings.go;l=459
```suggestion
	return strings.HasSuffix(name, suffixTotal)
```